gph/src/ripser.h

```diff
@@ -122,23 +123,23 @@ class binomial_coeff_table
     /* Transposed binomial table
      * It's transposed because access where done over the rows and not the
      * columns */
-    binomial_coeff_table(index_t k, index_t n) : B(n + 1, row_bc(k + 1, 0))
+    binomial_coeff_table(index_t n, index_t k) : B(k + 1, row_bc(n + 1, 0))
     {
-        for (index_t i = 0; i <= k; ++i) {
+        for (index_t i = 0; i <= n; ++i) {
             B[0][i] = 1;
-            if (i <= n)
+            if (i <= k)
                 B[i][i] = 1;
-            for (index_t j = 1; j < std::min(i, n + 1); ++j) {
+            for (index_t j = 1; j < std::min(i, k + 1); ++j) {
                 B[j][i] = B[j - 1][i - 1] + B[j][i - 1];
             }
-            check_overflow(B[std::min(i >> 1, n)][i]);
+            check_overflow(B[std::min(i >> 1, k)][i]);
         }
     }
 
-    index_t operator()(index_t k, index_t n) const
+    index_t operator()(index_t n, index_t k) const
     {
-        assert(n < B.size() && k < B[n].size() && n >= k - 1);
-        return B[n][k];
+        assert(k < B.size() && k < B[k].size() && n >= k - 1);
+        return B[k][n];
```

Comment:
Could we please also rename the `i`s to `j`s and vice-versa? That way, `j` is a column index and `i` is a row index, which I think is more standard than the other way around.

Reply:
Well, by convention in programming, we start with letter `i` in nested for loops and then we go by `j`, etc.
I think if the name of the indices are important, then we should rename then `idx_row` and `idx_col`, because otherwise in my opinion it is confusing.

Reply:
I do think the code is better understood by others if we swap the names, even if this breaks the nested for-loop convention.  Which `idx_row` and `idx_col` are you referring to?

Reply:
I understand the change, I still think it is not a good idea to swap `i` and `j` as names for the nested loops.
What I was suggesting is that if those name are relevant, instead of just being loops indices, we can replace as follow:
* `j` => `idx_col`
* `i` => `idx_row`

Otherwise, I am not favorable for the requested change, I think this will be confusing for anyone looking at the code to have loop using `i` as index name be inside of loop using `j` as name.

Reply:
Ah I see!  Well I guess we have different perspectives on this.  Those new names are a bit verbose, which I think impacts readability badly in another way.  And I can't think of better ones (I guess `r` and `c` are off the table?).  I'm happy to give this one up!

Reply:
It happens :)

On contrary, `r` and `c` are good candidates. They are not usual names for loops indices, I am definitely fine with this change. If I am right, I should proceed with the following change:
* `j` => `c`
* `i` => `r`

Right ? If yes, I will add just a short comment to indicate that `c` is for column index and `r` is for row index.

Reply:
Thinking about it a bit more, I think the next best option is to keep the names as they are but add a short comment saying that `B[j][i] = binom(i, j)`.

Reply:
(This comment can be added to the pre-existing comment above, on transposition.)

Reply:
I will add it then !
Thank you for the suggestion @ulupo and I sorry to be a bit stubborn

Reply:
Comment added, if fine for you, I will let you resolve the conversation

Reply:
Great, saw it, thanks!